tree.go

```diff
@@ -552,24 +570,105 @@ func (n *InternalNode) Commitment() *Point {
 }
 
 func (n *InternalNode) Commit() *Point {
-	poly := make([]Fr, NodeWidth)
-	emptyChildren := 256
-
+	if n.depth <= 1 {
+		return n.commitRoot()
+	}
 	if len(n.cow) != 0 {
```

Comment:
At this line 576, we do serial execution of work as before. If we're at these layers, we're already in separate goroutines that were fired in the top 2 layers.

Still, I did other optimizations explained right after this comment.